pulp_smash/utils.py

```diff
@@ -80,3 +86,48 @@ def poll_task(server_config, href):
         # This approach is dumb, in that we don't account for time spent
         # waiting for the Pulp server to respond to us.
         sleep(5)
+
+
+def get_broker(server_config):
+    """Build an object for managing the target system's AMQP broker.
+
+    Talk to the host named by ``server_config`` and use simple heuristics to
+    determine which AMQP broker is installed. If Qpid or RabbitMQ appear to be
+    installed, return a :class:`pulp_smash.cli.Service` object for managing
+    those services respectively. Otherwise, raise an exception.
+
+    Another option for solving this problem is to create a subclass of
+    ``Service``, like so:
+
+    >>> class Broker(Service):
+    ...     def __init__(self, server_config):
+    ...         service = …  # 'qpidd', 'rabbitmq', etc.
+    ...         super(Broker, self).__init__(server_config, service)
+
+    However, this approach has two drawbacks:
+
+    * It favors inheritance over composition.
+    * It violates the Liskov substitution principle, due to its ``__init__``
+      method requiring a more specific set of arguments.
+
+    :param pulp_smash.config.ServerConfig server_config: Information about the
+        system on which an AMQP broker exists.
+    :rtype: pulp_smash.cli.Service
+    """
+    # On Fedora 23, /usr/sbin and /usr/local/sbin are only added to the $PATH
+    # for login shells. (See /etc/profile.) As a result, logging into a system
+    # and executing `which qpidd` and remotely executing `ssh pulp.example.com
+    # which qpidd` may return different results. For that reason, we try both
+    # checks here.
+    client = cli.Client(server_config, cli.echo_handler)
+    executables = ('qpidd', 'rabbitmq')  # ordering indicates preference
+    for executable in executables:
+        file_check = ('[', '-f', '/usr/sbin/' + executable, ']')
+        if (client.run(('which', executable)).returncode == 0 or
```

Comment:
There is very little chance that the `qpidd` and `rabbitmq` binaries will move. Consider using:
`test -e /usr/sbin/qpidd` as a simple way to do this. I'm not sure `which` is providing much benefit here and it's introducing complexity.